### PR TITLE
Fix #10026

### PR DESCRIPTION
**Pull request by pengzhao-intel**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by marcoabreu** [comment]
Hi, I'll wait for Sheng or somebody equivalent to review this. Please hold on

**Comment by pengzhao-intel** [comment]
No problem. Thanks for the help.

File: python/mxnet/gluon/nn/basic_layers.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def __init__(self, **kwargs):
         super(Flatten, self).__init__(**kwargs)
 
     def hybrid_forward(self, F, x):
-        return x.reshape((0, -1))
+        return F.Flatten(x)
```
Comment (szha) [inline python/mxnet/gluon/nn/basic_layers.py:408]: This is not equivalent. https://mxnet.incubator.apache.org/versions/master/api/python/ndarray/ndarray.html?highlight=reshape#mxnet.ndarray.reshape

Comment (pengzhao-intel) [reply python/mxnet/gluon/nn/basic_layers.py:408]: @szha, thanks for the comments. 

I am not very familiar with the difference between gluon and python model.
But I see the `Flatten` is used in the AlexNext (googlenet, etc) under python example.  
https://github.com/apache/incubator-mxnet/blob/0ca6dd29cb923c59925c76981bfd5a632a72a8ca/example/image-classification/symbols/alexnet.py#L55

And other symbolic operators are used in gluon model as well, such as `BatchNorm`:
https://github.com/apache/incubator-mxnet/blob/0ca6dd29cb923c59925c76981bfd5a632a72a8ca/python/mxnet/gluon/nn/basic_layers.py#L341

What's functionality missing in `Flatten` operator so that it can't be used in gluon `Flatten` class?

**Comment by szha** [comment]
Non-equivalence. Please look up or try what the existing implementation does, and what Flatten op does.

File: python/mxnet/gluon/nn/basic_layers.py (hunk 1 above)
Comment (szha) [reply python/mxnet/gluon/nn/basic_layers.py:408]: Non-equivalence. Please look up or try what the existing implementation does, and what Flatten op does.

Comment (pengzhao-intel) [reply python/mxnet/gluon/nn/basic_layers.py:408]: Thanks. Got it and will keep it as-is.

**Comment by szha** [comment]
Is this still relevant?

**Comment by pengzhao-intel** [comment]
close this one. The code has been changed a lot and the original PR has been closed. 
Will submit the PR separately.
